### DISCUSSION TITLE
Water level sensor

### DISCUSSION THREAD

**Discussion by orrious** [tagyoureit/nodejs-poolController#542] [REM]
Has anyone implemented a water lever sensor?  I have room in my autofill and have been looking for something suitable.

**Comment by rstrouse** [comment]
I haven't seen anything that does a good job at that.   The ones that are used in fluid tanks use a form of radio reflection and do better at measuring the level of your wallet.  The tape level sensors are a novelty and the float based ones are simply indicate whether it is low.

**Reply by orrious** [reply]
The ultrasonic ones, I think that’s what you’re taking about, have zero resolution for the first 20cm. Which makes them no good, I tried. 

Tape I wouldn’t even consider

Floats are a set point as you’ve stated. That’s why I asked if anyone had something else they’ve used successfully

**Comment by orrious** [comment]
@rstrouse What are you thoughts on this?

https://www.amazon.com/dp/B099J6WQLW

**Reply by rstrouse** [reply]
The only thing that gives me pause with these are the resolution on the sending unit.  These have 12 resistance points,  so at 125mm you only get 10mm resolution.  It might give you the resolution you need.  I hope you are planning on a wet well for installation a skimmer just won't do.

**Reply by orrious** [reply]
I had planned on putting it in my auto fill.  https://www.amazon.com/Pentair-T40FW-Autofill-Automatic-Fluidmaster/dp/B004VU8U52
When I built the pool, I ran conduit to it in anticipation I might be able to use it to track the level of the water.  

Btw, good catch, I didn't see that the increments were so coarse.

**Comment by buvC8Q4z** [comment]
@orrious Wondering if you came up with a solution you're happy with for your pool water level sensor? I'm just starting to plan my automation as our pool remodelers show up tomorrow to start demolition, so trying to think ahead for laying wiring, conduit, etc. and was thinking of having them install a run of PVC from the pool wall and elbow up near the pool equipment, that I could then use for checking the pool water level, but want to figure out what sensor to use. Thank you.

**Reply by buvC8Q4z** [reply]
Hey @buvC8Q4z - I'm in the exact same position, but about a week behind you (demolition started yesterday). I've got the same thought regarding a PVC conduit below the water line and then elbow up near my pool equipment and controller. The only article I'd found was [this one](https://www.hackster.io/user3424878278/pool-fill-control-119ab7) from Instructables. In the article he walks through some water level sensors and recommended one of [these](https://www.amazon.com/Elecall-Stainless-Sensor-Switch-Floater/dp/B01M0I5YAL?th=1). At this point the important thing I needed to understand were the dimensions of the sensor so I'd know what size pipe to have, and it looks like I'll need 1 1/2" pipe for the sensor to be able to fit. That's as far as I've gotten - I'm still searching around for more information on pool water level automation.

**Reply by orrious** [reply]
I've been busy with doing a Solar Pergola project, haven't made my way back to this in a bit.  I do saltwater aquariums as well and noticed that Neptune Systems came out with a Liquid Level Sensor (LLS).  https://www.neptunesystems.com/fmm-accessories/

I assume it's the same tech as this, already packaged. https://www.adafruit.com/product/1786.  

This technology is what I'll be going with, more than likely.  It should be simple enough to read the resistance with an ADC.  

Btw, I wouldn't run a long pipe from your pool to your pad for this use.  You'll end up with stagnant water as the water in the pipe will never flow.  Try to have this as close to the pool as possible.  An autofill is a great location.

**Reply by buvC8Q4z** [reply]
Ahhh, good thought regarding the potential for stagnant water. My pipe should only be a few feet, but maybe that's still too far - have to rethink this. I wonder if I can include a bleed line to provide some level of water flow. Or, since my pool deck is being removed, install the pipe right next to the pool and then just pull the wiring to the sensor over to the pool equipment pad. More stuff to figure out. ;)

Regarding the sensors you linked....my concern is that (as the Instructables article mentioned) the author found these end up getting moisture inside (or at least the similar product the author had used).

**Reply by orrious** [reply]
I'm sure some implementations are better than others.  Some have a Teflon jacket, others have a plastic casing.  https://www.adafruit.com/product/2656

If your pool deck is up, do you have an auto fill / auto drain installed currently?  If not, look into doing so.  This is where I'm installing the E-tape sensor.  

https://www.amazon.com/Pentair-T40FW-Autofill-Automatic-Fluidmaster/dp/B004VU8U52

The Pentair one I've shown uses a toilet float to automatically fill the pool.  This is nice when it comes to replacement parts.  You will need to run water supply to it, a drain from it, and then another conduit for the sensor wire.

**Reply by buvC8Q4z** [reply]
Just finished talking with the plumbing guy.... Yeah the idea of the pipe over to the pool equipment to check the level is a bad idea, on multiple fronts. He helped me recognize that just installing a pool filler like the link you provided, was way simpler, and I'll install it right where the current fill water source/line is for the pool, so all I need is to dig a hole for the plastic housing and get a hole drilled in from the side of the pool to this housing. The $150 price tag on these is crazy for a plastic housing that contains a $10 Fluidmaster toilet fill valve....then again, the pool contractor wanted $750 to install one of these. =8O

**Reply by orrious** [reply]
It's not just the labor, its the liability if it leaks or has a problem.  I'm sure if you install it, he'll have you sign a waiver.
